apps/jubmoji-quest/src/pages/jubmojis/index.tsx

```diff
@@ -73,56 +159,59 @@ export default function JubmojisPage() {
+        <div
+          className={cn("grid  justify-between gap-2", {
+            "grid-cols-[1fr_70px]": isSearchMode,
+            "grid-cols-[1fr_120px]": !isSearchMode,
+          })}
+        >
+          <Input
+            type="search"
+            placeholder="Search your private collection"
+            value={search}
+            onChange={onSearch}
+          />
+          {isSearchMode ? (
+            <button
+              onClick={() => setIsSearchMode(false)}
+              className="font-dm-sans text-[13px]"
+            >
+              Cancel
+            </button>
+          ) : (
+            <Button
+              icon={<Icons.download className="text-black" />}
+              size="sm"
+              variant="blue"
+              onClick={() => setBackupModalOpen(true)}
+            >
+              Back up!
+            </Button>
+          )}
+        </div>
 
-          // TODO: CHANGE this in a better way without loop
-          if (selectedJubmoji.sig !== jubmoji.sig) return null;
+        {isSearchMode ? <JubmojiSearchItems /> : <JubmojiContent />}
 
-          // Emoji is fixed in hardware and fetched from hardcoded card metadata file
-          const { emoji } = cardPubKeys[jubmoji.pubKeyIndex];
-          // Name, owner, and collectsFor are set by the current cardholder and fetched from the backend
-          const { name, owner, collectsFor } =
-            jubmojiCollectionCards[jubmoji.pubKeyIndex];
-          // Image path is fetched from the hardcoded card image map for offline use
-          const imagePath = cardPubKeys[jubmoji.pubKeyIndex].imageBlobUrl;
+        {!isSearchMode && (
+          <div className="mt-auto">
+            <JubmojiNavWrapper>
+              {collectedJubmojis?.map((jubmoji, index) => {
+                const isActive = jubmoji?.pubKeyIndex === selectedPubKeyIndex;
 
-          return (
-            <div key={i} className="my-4">
-              <CollectionCard
-                label={name}
-                icon={emoji}
-                edition={jubmoji.msgNonce}
-                owner={owner}
-                cardBackImage={imagePath}
-                actions={<span>backed up</span>}
-                quests={collectsFor}
-              />
-            </div>
-          );
-        })}
-        <div className="mt-auto">
-          <JubmojiNavWrapper>
-            {jubmojis?.map((jubmoji, index) => {
-              const isActive = index === active;
-              const { emoji } = cardPubKeys[jubmoji.pubKeyIndex];
-
-              return (
-                <JubmojiNavItem
-                  key={index}
-                  active={isActive}
-                  onClick={() => onSelectJubmoji(jubmoji, index)}
-                >
-                  {emoji}
-                </JubmojiNavItem>
-              );
-            })}
-          </JubmojiNavWrapper>
-        </div>
+                if (!jubmoji) return null;
+                return (
+                  <JubmojiNavItem
```

Comment:
![IMG_6835](https://github.com/jubmoji/jubmoji.quest/assets/6290822/e1b83ad6-1259-41c3-be39-1f7d67574de8)

Right now, edition number isn't properly displaying. There are also two quest buttons that occasionally pop up? It seems like when a jubmoji actually has >0 quests there are 2 quests buttons lol

Reply:
@vb7401  Oh! thk!